src/hypofuzz/dashboard/dashboard.py

```diff
@@ -41,6 +42,16 @@
 websockets: set["HypofuzzWebsocket"] = set()
 
 
+class DashboardEventType(IntEnum):
+    # minimize header frame overhead with a shared IntEnum definition between
+    # python and ts.
+    ADD_TESTS = 1
+    ADD_REPORTS = 2
+    ADD_ROLLING_OBSERVATIONS = 3
+    ADD_CORPUS_OBSERVATIONS = 4
+    SET_FAILURE = 5
+
+
 def test_for_websocket(test: Test) -> dict[str, Any]:
     # Limit to 1000 reports per test. If we're over 1000 total reports, sample
     # evenly from each worker.
```

Comment:
Let's also adjust the slicing logic to ensure that the last few (20?  at least one) are also included - it'd be sad to miss a breakthrough just because of our display logic.

Reply:
first and last 10 are now kept, and those 20 total don't count towards the soft limit of 1000